reccmp/isledecomp/compare/core.py

```diff
@@ -830,7 +831,27 @@ def _compare_match(self, match: ReccmpMatch) -> DiffReport | None:
             )
 
         if match.entity_type == EntityType.FUNCTION:
-            return self.function_comparator.compare_function(match)
+            best_name = match.best_name()
+            assert best_name is not None
+
+            diff_result = self.function_comparator.compare_function(match)
+            if diff_result.match_ratio != 1.0:
+                grouped_codes = list(get_grouped_opcodes(diff_result.codes, n=10))
+                udiff = combined_diff(
+                    grouped_codes, diff_result.orig_inst, diff_result.recomp_inst
+                )
+            else:
+                udiff = None
+
+            return DiffReport(
+                match_type=EntityType.FUNCTION,
+                orig_addr=match.orig_addr,
+                recomp_addr=match.recomp_addr,
+                name=best_name,
+                udiff=udiff,
+                ratio=diff_result.match_ratio,
+                is_effective_match=diff_result.is_effective_match,
+            )
```

Comment:
What I don't quite like is the size increase of `core.py` which I'd like to keep smaller. Maybe we pull this out into a helper function? Doesn't have to move back into the `function_comparator`